rest_framework/fields.py

```diff
@@ -939,7 +945,12 @@ def __init__(self, **kwargs):
 
     def to_internal_value(self, data):
 
-        if isinstance(data, six.text_type) and len(data) > self.MAX_STRING_LENGTH:
+        data = smart_text(data).strip()
```

Comment:
Not keen on this. For the default behavior we'd now be unnecessarily transforming the incoming value into a string, and then back to a float again. Perhaps we could push this inside the `if self.localize:` block?